config/network/config.go

```diff
@@ -62,16 +77,49 @@ type UnicastRateLimitersConfig struct {
 	// Recommended size is 10 * number of authorized nodes to allow for churn.
-	SpamRecordCacheSize uint32 `mapstructure:"spam-record-cache-size"`
+	SpamRecordCacheSize uint32 `mapstructure:"alsp-spam-record-cache-size"`
 
 	// SpamReportQueueSize is the size of the queue for spam records. The queue is used to store spam records
 	// temporarily till they are picked by the workers. When the queue is full, new spam records are dropped.
 	// Recommended size is 100 * number of authorized nodes to allow for churn.
-	SpamReportQueueSize uint32 `mapstructure:"spam-report-queue-size"`
+	SpamReportQueueSize uint32 `mapstructure:"alsp-spam-report-queue-size"`
 
 	// DisablePenalty indicates whether applying the penalty to the misbehaving node is disabled.
 	// When disabled, the ALSP module logs the misbehavior reports and updates the metrics, but does not apply the penalty.
 	// This is useful for managing production incidents.
 	// Note: under normal circumstances, the ALSP module should not be disabled.
-	DisablePenalty bool `mapstructure:"disable-penalty"`
+	DisablePenalty bool `mapstructure:"alsp-disable-penalty"`
+}
+
+// SetAliases this func sets an aliases for each CLI flag defined for network config overrides to it's corresponding
+// full key in the viper config store. This is required because in our config.yml file all configuration values for the
+// Flow network are stored one level down on the network-config property. When the default config is bootstrapped viper will
+// store these values with the "network-config." prefix on the config key, because we do not want to use CLI flags like --network-config.networking-connection-pruning
+// to override default values we instead use cleans flags like --networking-connection-pruning and create an alias from networking-connection-pruning -> network-config.networking-connection-pruning
+// to ensure overrides happen as expected.
+// Args:
+// *viper.Viper: instance of the viper store to register network config aliases on.
+// Returns:
+// error: if a flag does not have a corresponding key in the viper store.
+func SetAliases(conf *viper.Viper) error {
+	m := make(map[string]string)
+	// create map of key -> full pathkey
+	// ie: "networking-connection-pruning" -> "network-config.networking-connection-pruning"
+	for _, key := range conf.AllKeys() {
+		s := strings.Split(key, ".")
+		// check len of s, we expect all network keys to have a single prefix "network-config"
+		// s should always contain only 2 elements
+		if len(s) == 2 {
+			m[s[1]] = key
+		}
+	}
+	// each flag name should correspond to exactly one key in our config store after it is loaded with the default config
+	for _, flagName := range AllFlagNames() {
+		fullKey, ok := m[flagName]
+		if !ok {
+			return fmt.Errorf("invalid network configuration missing configuration key flag name %s check config file and cli flags", flagName)
+		}
+		conf.RegisterAlias(fullKey, flagName)
```

Comment:
shouldn't this be `conf.RegisterAlias(flagName, fullKey)` according to the documentation of Viper?

Reply:
No the usage is correct. When keys bound from the CLI are added to the config store they are added without any of the prefixes. Because we want to access config values without using long keys with prefixes we set an alias from the keys bound from the CLI -> keys read in from config files that may have prefixes.